plugins/providers/virtualbox/action.rb

```diff
@@ -79,13 +79,14 @@ def self.action_boot
           b.use ForwardPorts
           b.use SetHostname
           b.use SaneDefaults
-          b.use CloudInitSetup
```

Comment:
The need for cloud init can be set into the env in the `.action_up`. Here in the `.action_boot` the `IsEnvSet` action can be used to check for it, and then add the cloud init related actions if needed.

```ruby
b.use Call, IsEnvSet, :cloud_init do |env, b2|
  if env[:result]
    b2.use CloudInitSetup
  end
  b2.use CleanupDisks
  b2.use Disk
  ...
  b2.use WaitForCommunicator, [:starting, :running]
  if env[:result]
    b2.use CloudInitWait
  end
  b2.use Customize, "post-comm"
  b2.use CheckGuestAdditions
end
```